### DISCUSSION TITLE
GitHub serving old page

### DISCUSSION THREAD

**Discussion by manas96** [community/community#49753] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I decided to overhaul my website and started learning Hugo to use as a static site generator. I want a page with a photo gallery. 
At some point in the past, I used a Hugo theme to serve the page at [https://manas96.github.io/photography/](https://manas96.github.io/photography/). I just used the default theme that redirects to the creator's webpage:
![Screenshot 2023-03-10 202623](https://user-images.githubusercontent.com/11874996/224457316-90f93078-66b7-41bd-a0a9-037b7a36cd6e.png)

Now, I have made a new `/photography` page that renders like this on my local machine:
![Screenshot 2023-03-10 202835](https://user-images.githubusercontent.com/11874996/224457443-58c70f3f-dfb3-41c4-a086-375f6651e753.png)

I have pushed these changes to github pages repo and the build task has succeeded. However, at the time of writing this post if I visit  https://manas96.github.io/photography/ , I still get screenshot 1. In an effort to check if I accidentally left something in my build, I have searched for "Ram Patra" all over my github page repository, but can't find any text matching that. 
So my conclusion is GitHub pages have not refreshed their cache of `/photography` somehow?
What should I do to get my latest changes (screenshot 2) to be served?

Thanks!

**Comment by itspasindu** [comment, marked as answer]
It's possible that your changes have not yet been reflected on the live website due to caching. Here are some steps you can take to ensure that your latest changes are served:

    Clear your browser cache: Try clearing your browser cache and refreshing the page. This will force your browser to download the latest version of the page.

    Wait for the cache to expire: GitHub Pages caches static content for up to 10 minutes by default. If you wait for this time to expire, your changes should be reflected on the live website.

    Disable GitHub Pages caching: You can try disabling GitHub Pages caching altogether by adding a no-cache HTTP header to your website. To do this, create a file named _headers in the root directory of your website and add the following line to it:

    yaml

```
/photography/* 
Cache-Control: no-cache
```

This will tell GitHub Pages not to cache any content under the /photography directory.

Force a rebuild of your website: You can try forcing a rebuild of your website by pushing an empty commit to your repository. This will trigger a new build of your website and should force GitHub Pages to serve the latest version of your site. To do this, simply run the following commands:

perl

  ```
  git commit --allow-empty -m "force rebuild"
    git push
```

    This will push an empty commit to your repository and trigger a new build of your website.

Hopefully, one of these steps will help you get your latest changes to be served. If none of these steps work, you may need to reach out to GitHub support for further assistance.

**Reply by manas96** [reply]
Thanks for the suggestions, I tried all of them but still have the same issue.
To be clear, is the file name `_headers` or `_headers.yaml`?

**Reply by manas96** [reply]
I solved this, the culprit was that I had another repository that had published to this link. I didn't know multiple repositories could publish to the same baseurl, so I didn't check until it was pointed out to me by github support. 

Thanks for the help!

**Reply by ManuelLevi** [reply]
Hi @manas96 , I'm having the same problem.

What was the solution? Did you find another repository that was also publishing to this baseurl and you had to remove it?
How did you find this repo?

Thank you, ML

**Reply by manas96** [reply]
Hi @ManuelLevi the solution was that I removed the older repo that was publishing to this link. I found it because it was my own repo.

**Reply by vincerubinetti** [reply]
> Disable GitHub Pages caching: You can try disabling GitHub Pages caching altogether by adding a no-cache HTTP header to your website. To do this, create a file named _headers in the root directory of your website and add the following line to it:
> /photography/* 
> Cache-Control: no-cache
> This will tell GitHub Pages not to cache any content under the /photography directory.

What is your basis for this? This reads like an AI hallucination. There's no indication that GitHub Pages allows setting HTTP Headers in any kind of general way like this.

https://github.com/orgs/community/discussions/54257

**Comment by Vacheslav909** [comment]
Не чистий не має більше влади.Замість нього Вячеслав Іванович. Новий генеральний директор та власник Google, Play Market, всіх світових серверів та машини часу Йовжій Вячеслав Іванович передай іншим +380916112825 slavik19950527909@gmail.com slavik19950527@mail.ru +380982614680 viber

**Comment by Vacheslav909** [comment]
0916112825 telegram

сб, 11 бер. 2023, 19:07 користувач Manas Kale ***@***.***>
пише:

**Comment by ManuelLevi** [comment]
Thanks!
I had another problem. When I committed and pushed the code I didn't notice
I pushed "public" as well. So while the source was not there these ones
were being served.
Took me a while to find the problem. Hope this helps someone else.

**Reply by lewybagz** [reply]
I'm having this problem as well. I pushed public but the only file in there is vite.svg so i doubt thats my problem. i have tried all of the above suggested solutions

**Comment by yairEO** [comment]
### Go to "Actions" tab and click the latest "*pages build and deployment*" workflow:

<kbd><img src="https://github.com/user-attachments/assets/0a904436-43d3-4038-901b-df8c25cecec7" /></kbd>

### And re-run:

<kbd><img src="https://github.com/user-attachments/assets/4fc6c9f3-30ed-4c06-b6c2-9cb72cd098bd" /></kbd>
